### PR TITLE
Python3 updates

### PR DESCRIPTION
**Pull request by ricardoamador**
Python 2 has been deprecated in the use of recipes so we need to port python 2 methods to python 3.

*List which issues are fixed by this PR. You must list at least one issue.*
This is related to this CL with the main fixes: https://flutter-review.git.corp.google.com/c/recipes/+/38148

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: ci/firebase_testlab.py [hunk 1]
```diff
@@ -64,9 +66,11 @@ def run_firebase_test(apk, results_dir):
 
 def check_logcat(results_dir):
   logcat = subprocess.check_output([
-      'gsutil', 'cat',
+      'gsutil', 
+      'cat',
       '%s/%s/*/logcat' % (BUCKET, results_dir)
-  ])
+  ]).decode(encoding)
```
Comment (godofredoc) [inline ci/firebase_testlab.py:72]: Seems like we'll need some logic to support python2 and 3 probably.

File: ci/firebase_testlab.py [hunk 2]
```diff
@@ -66,7 +68,7 @@ def check_logcat(results_dir):
   logcat = subprocess.check_output([
       'gsutil', 'cat',
       '%s/%s/*/logcat' % (BUCKET, results_dir)
-  ])
+  ]).decode(ENCODING)
```
Comment (godofredoc) [inline ci/firebase_testlab.py:71]: Let's try with:

```
logcat = subprocess.check_output([ ....
logcat = logcat if isinstance(logcat, str) else logcat.decode(ENCODING)
```

Comment (ricardoamador) [reply ci/firebase_testlab.py:71]: Okay no problem but I thought the exercise was that we were no longer targeting python 2?

Comment (ricardoamador) [reply ci/firebase_testlab.py:71]: Okay gotcha. I though we going full on to python 3. Let me make the changes.

**Comment by ricardoamador** [comment]
I'm not sure what that formatting check is for or how to fix it. It is judging lines that I had not modified and odd spaces are appearing suggesting that I have modified the code.

**Comment by godofredoc** [comment]
> I'm not sure what that formatting check is for or how to fix it. It is judging lines that I had not modified and odd spaces are appearing suggesting that I have modified the code.

Yeah, the linting of engine repo is weird if you are not following the engine workflow. I'd recommend to setup your engine environment using steps 5, 6 and 7 of https://github.com/flutter/flutter/wiki/Setting-up-the-Engine-development-environment. That will register a hook that will validate your changes before uploading.

**Comment by ricardoamador** [comment]
Engine is a funky environment for git so I need to recreate this.

**Pull request closed without merging**